fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/domain/SearchQueryRenderer.java

```diff
@@ -1055,7 +1056,32 @@ protected WhereFragment getQuantityFilter(QueryParameter queryParm, String param
     protected WhereFragment getDateFilter(QueryParameter queryParm, String paramAlias) {
         WhereFragment where = new WhereFragment();
         NewDateParmBehaviorUtil util = new NewDateParmBehaviorUtil();
-        util.executeBehavior(where, queryParm, paramAlias);
+        
+        // It is possible that multiple date parameters could be chained together if there were
+        // multiple query parameters specified for the same date search parameter and we were
+        // able to consolidate them. Check specifically if we have a consolidated parameter which
+        // specifies a range, and if so, build a range filter.
+        if (queryParm.isChained() && queryParm.getChain().size() == 1) {
+            Prefix prefix1 = queryParm.getValues().get(0).getPrefix();
+            Prefix prefix2 = queryParm.getNextParameter().getValues().get(0).getPrefix();
+            if ((Prefix.GT.equals(prefix1) || Prefix.GE.equals(prefix1) || Prefix.SA.equals(prefix1)) &&
```

Comment:
Could prefix1 have the LT/LE/EB and prefix2 have the GT/GE/SA, or is that permutation not possible?

Reply:
Ah, ok. The consolidateDateParms() method explicitly puts them in that order.  Might be good to handle either order so the two methods are not assuming order; or at least document in consolidateDateParms() that getDateFilter assumes the order.

Reply:
updated to handle either order